src/app/loadout-builder/item-filter.ts

```diff
@@ -58,7 +58,10 @@ export function filterItems(
     if (filteredItems[bucket]) {
       filteredItems[bucket] = filteredItems[bucket].filter(
         (item) =>
-          (bucket !== lockedExotic?.bucketHash || item.hash === lockedExotic.def.hash) &&
+          (!lockedExotic ||
+            (bucket === lockedExotic.bucketHash ?
+              item.hash === lockedExotic.def.hash :
+              item.equippingLabel !== lockedExotic.def.equippingBlock!.uniqueLabel)) &&
```

Comment:
Since we can only ever have one exotic in set we may as well ignore all exotics in other buckets. We identify exotics by if they have an equipping label, so this will do it.

```suggestion
              !item.equippingLabel)) &&
```

Reply:
I did it this way because technically Bungie says that only items with the equal equipping labels can't be equipped at the same time. Right now there is only "exotic_armor" but if there were other ones this would handle them correctly.

Reply:
Also the checks in process.ts are still necessary because we might not have an exotic locked at all, and then we still need to check all combinations and filter the double exotic ones.

Reply:
Looking around, other places like LoadoutDrawer.tsx and loadout-utils.ts already handle equipping_label with equality checks, so its really just process.ts that is doing the non-null check. I could change process.ts to use equality checks as well to fix it if you'd like.

Reply:
> Looking around, other places like LoadoutDrawer.tsx and loadout-utils.ts already handle equipping_label with equality checks, so its really just process.ts that is doing the non-null check. I could change process.ts to use equality checks as well to fix it if you'd like.

They handle equipping label with equality checks as there are currently two equipping labels I am aware of, one for weapons and one for armour (exotic_weapon and exotic_armor I think), so they need to handle them with equality checks as both types can appear in a loadout.

Reply:
I'd rather keep the equality check since its more semantically correct (and I pushed another commit that does the same for process.ts), but I can totally change the non-null assertion to check equippable first (or add that check to ExoticPicker.tsx). Would that work for you?

Reply:
Yeah all good with leaving the equality check in as it is more correct as you said.

I've actually just gone and done some digging and I think I am being unnecessarily cautious. We are building `item.equippingLabel` from the def so we would need a mismatch between manifest defs and items in the store which should never happen. See https://github.com/DestinyItemManager/DIM/blob/master/src/app/inventory/store/d2-item-factory.ts#L381. Going to approve and merge as is.